### DISCUSSION TITLE
JSX code block does not highlight the code

### DISCUSSION THREAD

**Discussion by JakeBartoletta** [squidfunk/mkdocs-material#6219] [Q&A] [answered]
Hello all,

I am currently running into an issue where if i set a code block to `jsx` it does not highlight the code at all. I can set other languages like `js` or `html`. I have gone through the documentation to see if i am missing anything but no luck. I also changed my theme to mkdocs and it seems to work on that theme setting. Anyone know how to fix for this theme?

Here are some screenshots

Material theme:
![image](https://github.com/squidfunk/mkdocs-material/assets/102685248/dc8977e9-90c7-4e21-8251-b594bfad05aa)

Mkdocs theme:
![image](https://github.com/squidfunk/mkdocs-material/assets/102685248/02bde8c8-91c5-4b7d-a48f-5143864dabd5)

**Comment by alexvoss** [comment]
Which of the possible [highlighters](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown-extensions#highlight) are you using?

**Reply by JakeBartoletta** [reply]
Using Pygments i believe. Here is what i have in my markdown extensions

![image](https://github.com/squidfunk/mkdocs-material/assets/102685248/7daefda0-492d-4b86-94e6-eec6834a3258)

**Reply by alexvoss** [reply]
It looks like 'raw' MkDocs uses [highlight.js](https://highlightjs.org/), while I assume you would be using [Pygments](https://pygments.org/)? Pygments does have a JSX lexer, so the question would be why it does not get applied.

**Reply by JakeBartoletta** [reply]
Not sure, I have been looking at [these](https://facelessuser.github.io/pymdown-extensions/extensions/highlight/) docs but seems like it only works for some languages.

**Reply by alexvoss** [reply]
Can you produce a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) so we can be sure we are looking at the same thing when trying to find out what is happening? Please make sure to include only the minimum files, no customizations and a minimal `mkdocs.yml` so we can really narrow this down.

**Reply by facelessuser** [reply]
From what I understand, you should probably use https://github.com/fcurella/jsx-lexer.

**Reply by alexvoss** [reply, marked as answer]
It would seem that [this lexer was recently added into Pygments](https://github.com/pygments/pygments/pull/2524), so a version bump would potentially solve this? That is, once Pygments releases this.

**Reply by facelessuser** [reply]
No, I'm on the lateset Pygments, `jsx`, `react`, none of them get applied. I found the referenced package in this thread: https://github.com/pygments/pygments/issues/709. It may be that it has been added recently, but the changelog doesn't mention it, so maybe it is erroneously in the docs and not actually present yet.

**Reply by kamilkrzyskow** [reply]
The version bump on the repository (aka new release) would fix it, the JSX support was added in September, whereas the 2.16.1 release is from August
I would also like to mention that using custom plugin lexers, introduces an issue with the lack of cache of such lexers, which increases build times.

**Reply by kamilkrzyskow** [reply]
You could also just run `pip install git+https://github.com/pygments/pygments.git` to install the latest version temporarily before the official release

**Reply by JakeBartoletta** [reply]
Just seen that on Pygments `2.17.0` they just added support for JSX lexer, I just assumed it was already added since its in the documentation.

**Reply by facelessuser** [reply]
It's likely not live yet as PyPI shows 2.16.1 as the latest. So maybe it will release very soon.

<img width="865" alt="Screenshot 2023-10-20 at 9 36 15 AM" src="https://github.com/squidfunk/mkdocs-material/assets/1055125/30008a99-7398-42e1-8536-268b38515590">

**Comment by JakeBartoletta** [comment]
> It's likely not live yet as PyPI shows 2.16.1 as the latest. So maybe it will release very soon.
> <img alt="Screenshot 2023-10-20 at 9 36 15 AM" width="865" src="https://user-images.githubusercontent.com/1055125/276979932-30008a99-7398-42e1-8536-268b38515590.png">

I hope so, they have their changelog on their site for v2.17.0

![image](https://github.com/squidfunk/mkdocs-material/assets/102685248/a2907895-71ab-4f8b-989a-2cceb866b0f8)

**Reply by facelessuser** [reply]
Yeah, but it says unreleased, but I imagine it will come soon-ish.
